src/python/pants/backend/go/goals/check.py

```diff
@@ -24,7 +24,7 @@ class GoCheckFieldSet(FieldSet):
 
 class GoCheckRequest(CheckRequest):
     field_set_type = GoCheckFieldSet
-    name = "go compile"
+    name = "go-compile"
```

Comment:
(Non-blocking, this PR is an improvement so it belongs in)
This is the odd duckling, which got me wondering what these are used for.

AFAICT (and `mypy` backs me up) they are only used on the superclass for `debug_hint`. That could probably get away with using `cls.__name__` :man_shrugging: . Then the only remaining usages are in each tool's `@rule`s as a DRY way of using the name (E.g. `formatter_name=request.name`). In that case, we could reference the `subsystem.options_scope` in all cases except this one, where we could hardcode it.

Of course, instead of an implicit convention (which this now has), in the future https://github.com/pantsbuild/pants/pull/14238 could implement a helper method which then allows for a codified convention :wink:

Reply:
See https://github.com/pantsbuild/pants/pull/14356, which this is all prework for.

Reply:
I see. It'd be nice if these things were more automagical if the client followed a convention (much like how Django operates for a lot of things). But I'm just soapboxing now :smile:

Reply:
I see. It'd be nice if these things were more automagical if the client followed a convention (much like how Django operates for a lot of things). 

But I'm just soapboxing now :smile: